### ISSUE TITLE
Problem with plugins in debug mode (wrong path to cfgs)

### ISSUE THREAD

**Issue by n0guest** [netdata/netdata#2593]
Hello.

I've tried to use v1.7.x and found some strange behavior. Before v1.7.x (e.g. v1.6.x or earlier) I used such command to debug plugins:

```
/usr/bin/python /usr/lib/x86_64-linux-gnu/netdata/plugins.d/python.d.plugin debug nginx 1
```

But after upgrading to v1.7.x I got this problem:

```
# /usr/bin/python /usr/lib/x86_64-linux-gnu/netdata/plugins.d/python.d.plugin debug 1 2>&1 | head -n 5
2017-08-13 22:49:43:  INFO: Using python v2
2017-08-13 22:49:43:  INFO: YAML output is ordered
2017-08-13 22:49:43: python.d INFO: reading configuration file: /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf
2017-08-13 22:49:43: python.d ERROR: /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf reading error: [Errno 2] No such file or directory: '/usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf'
2017-08-13 22:49:43: python.d INFO: MODULES_DIR='/usr/lib/x86_64-linux-gnu/netdata/python.d/', CONFIG_DIR='/usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=[]
```
As you can see, there is one missing reference to "parent dir" . I.e. instead of 4:

> /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf

You should have 5 (at least this is actual for Debian/Ubuntu) to build a "correct" path to conf file:

```
# ls -l /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf
ls: cannot access '/usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/python.d.conf': No such file or directory
# ls -l /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../../etc/netdata/python.d.conf
-rw-r--r-- 1 root root 1518 Aug 13 04:04 /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../../etc/netdata/python.d.conf
```
I also checked environment of netdata's child process and found this:

```
# cat /proc/4728/environ | xargs -0 -n 1 echo | grep -E '^NETDATA_CONFIG_DIR'
NETDATA_CONFIG_DIR=/etc/netdata
```
So, here is a workaround for this problem:

```
NETDATA_CONFIG_DIR=/etc/netdata /usr/bin/python /usr/lib/x86_64-linux-gnu/netdata/plugins.d/python.d.plugin debug nginx 1
```

Looks like current problem caused by this commit https://github.com/firehol/netdata/commit/30a4d477a9eea6d4b9d3d996fda74b94b0b0b3dc.

Is this a bug or i missed something?

Thanks.

**Comment by ktsaou** [comment]
Hi,

The default installation of netdata (ie via `netdata-installer.sh`) always installs netdata in a way that `../../../../etc/netdata` (4 parents) points to the netdata configuration directory.

I understand this is not always case when netdata is not installed with `netdata-installer.sh`.

So, the right way to do this, is to convert all plugins to `.in` and have the makefiles search/replace the installation directory to them.

**Comment by n0guest** [comment]
> I understand this is not always case when netdata is not installed with netdata-installer.sh.

Exactly. In my case netdata was installed via deb package, which was built with using of contrib/debian/rules with minor changes (disabled systemd for trusty, enabled nfacct plugin for trusty/xenial/jessie).

> So, the right way to do this, is to convert all plugins to .in and have the makefiles search/replace the installation directory to them.

Is this also actual for the scenario when installing from a package? As far as I know, paths are substituted somewhere under control of dpkg-buildpackage (configure+make+etc).

**Comment by Karolis92** [comment]
I get similar paths problem when trying to configure email notifications:
```
$ /usr/lib/x86_64-linux-gnu/netdata/plugins.d/alarm-notify.sh test

# SENDING TEST WARNING ALARM TO ROLE: sysadmin
2018-10-03 19:55:30: alarm-notify.sh: ERROR: Cannot find file /usr/lib/x86_64-linux-gnu/netdata/plugins.d/../../../../etc/netdata/health_alarm_notify.conf. Using internal defaults.
2018-10-03 19:55:31: alarm-notify.sh: INFO: sent email notification for: my_server test.chart.test_alarm is WARNING to 'root'
# OK
```
nedata installed with apt-get

**Comment by ktsaou** [comment]
Please update your netdata.
This has been fixed with #4182

**Comment by Karolis92** [comment]
@ktsaou, I am on the newest release:
```
$ sudo apt install -t unstable netdata
Reading package lists... Done
Building dependency tree       
Reading state information... Done
netdata is already the newest version (1.10.0+dfsg-3).
```

**Comment by ktsaou** [comment]
yes, I meant it is fixed in the git repo, so 1.11 will have it
